File: programmers/src/main/java/shane/leetcode/problems/easy/Q1528ShuffleString.java
package shane.leetcode.problems.easy;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

public class Q1528ShuffleString {

    @Test
    void test() {
        Assertions.assertThat(restoreString("codeleet", new int[]{4, 5, 6, 7, 0, 2, 1, 3})).isEqualTo("leetcode");
        Assertions.assertThat(restoreString("abc", new int[]{0, 1, 2})).isEqualTo("abc");
    }

    public String restoreString(String s, int[] indices) {
        char[] answer = new char[indices.length];
        for (int i = 0; i < indices.length; i++) {
            answer[indices[i]] = s.charAt(i);
        }

        return String.copyValueOf(answer);

    }
}

File: src/main/kotlin/shane/leetkode/problems/easy/Q1528ShuffleString.kt
package shane.leetkode.problems.easy

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Runtime: 277 ms, faster than 81.32% of Kotlin online submissions for Shuffle String.
 * Memory Usage: 38 MB, less than 94.51% of Kotlin online submissions for Shuffle String.
 */
class Q1528ShuffleString {

    @Test
    fun test() {
        Assertions.assertThat(restoreString("codeleet", intArrayOf(4, 5, 6, 7, 0, 2, 1, 3))).isEqualTo("leetcode")
    }

    fun restoreString(s: String, indices: IntArray): String {
        val arr = CharArray(s.length)
        for (i in s.indices) {
            arr[indices[i]] = s[i]
        }
        return String(arr)
    }
}
